webapp/src/components/Field/RecordList.vue

```diff
@@ -65,7 +83,7 @@ export default {
     const self = this;
     return {
       removals: 0,
-      types: ['A', 'AAAA', 'MX', 'NS', 'CNAME', 'TXT', 'SPF', 'CAA', 'TLSA', 'OPENPGPKEY', 'PTR', 'SRV', 'DS'],
+      types: ['A', 'AAAA', 'MX', 'NS', 'CNAME', 'TXT', 'SPF', 'CAA', 'TLSA', 'OPENPGPKEY', 'PTR', 'SRV', 'DS', 'Subnet'],
```

Comment:
is this list ever visible to the user?

Reply:
No, it's only an internal whitelist that's used inside `getRecordComponentName()` (in the same file)

Reply:
There is a side-effect that is, I think, not problematic. As before, a user can enter anything into the record type field (it's an editable drop-down). If a record type is entered that is not in the above list, a `GenericText` input field will be displayed for the value. Upon submission, the unknown record type is sent to the API along with the record value. The GUI doesn't need to know the semantics.

Accordingly, if the user now enters "Subnet" into the record type field, then the input field for the value will now be rendered from `Record/Subnet.vue` (because the above code change adds it to the list of known input types). The side-effect now is that such an RRset would include the input validation from the `RecordSubnet` component.

Fyi, the list displayed in the dropdown is defined here: https://github.com/desec-io/desec-stack/blob/4a143511d582fe54a3ab39d7880449a74817b983/webapp/src/components/Field/RRSetType.vue#L44-L57

Reply:
fair enough